### PR TITLE
adds dimensions for spartan-7 series FPGA BGA packages

### PR DESCRIPTION
**Pull request by herostrat**
This PR extends the previous PR with spartan-7 footprints.
https://github.com/pointhi/kicad-footprint-generator/pull/343#event-2308708917

Footprint PR: https://github.com/KiCad/kicad-footprints/pull/1568

### CODE REVIEW

File: scripts/Package_BGA/bga_xilinx.yml [hunk 1]
```diff
@@ -150,3 +150,60 @@ Xilinx_FFG1156:
...
+  pad_diameter: 0.275
+  mask_margin: 0.05
+  layout_x: 14
+  layout_y: 14
+Xilinx_FTGB196:
+  description: "Spartan-7 BGA, 14x14 grid, 13x13mm package, 1mm pitch; https://www.xilinx.com/support/documentation/user_guides/ug475_7Series_Pkg_Pinout.pdf#page=261, NSMD pad definition Appendix A"
+  pkg_width: 15
+  pkg_height: 15
+  pitch: 1
+  pad_diameter: 0.5
```
Comment (evanshultz) [inline scripts/Package_BGA/bga_xilinx.yml]: Should be 0.4

File: scripts/Package_BGA/bga_xilinx.yml [hunk 2]
```diff
@@ -150,3 +150,60 @@ Xilinx_FFG1156:
...
+  description: "Spartan-7 BGA, 14x14 grid, 8x8mm package, 0.5mm pitch; https://www.xilinx.com/support/documentation/user_guides/ug475_7Series_Pkg_Pinout.pdf#page=260, NSMD pad definition Appendix A"
+  pkg_width: 8
+  pkg_height: 8
+  pitch: 0.5
+  pad_diameter: 0.275
+  mask_margin: 0.05
+  layout_x: 14
+  layout_y: 14
+Xilinx_FTGB196:
+  description: "Spartan-7 BGA, 14x14 grid, 13x13mm package, 1mm pitch; https://www.xilinx.com/support/documentation/user_guides/ug475_7Series_Pkg_Pinout.pdf#page=261, NSMD pad definition Appendix A"
```
Comment (evanshultz) [inline scripts/Package_BGA/bga_xilinx.yml]: 15x15mm

File: scripts/Package_BGA/bga_xilinx.yml [hunk 3]
```diff
@@ -150,3 +150,60 @@ Xilinx_FFG1156:
   mask_margin: 0.05
   layout_x: 34
   layout_y: 34
+  
+# Spartan-7 FPGA Packages
+# Notes:
+Xilinx_CPGA196:
+  description: "Spartan-7 BGA, 14x14 grid, 8x8mm package, 0.5mm pitch; https://www.xilinx.com/support/documentation/user_guides/ug475_7Series_Pkg_Pinout.pdf#page=260, NSMD pad definition Appendix A"
+  pkg_width: 8
```
Comment (evanshultz) [inline scripts/Package_BGA/bga_xilinx.yml]: All are missing the additional_tags item which I thought we were going to put on every one, right? Did I misunderstand?

**Comment by evanshultz** [comment]
@herostrat 
I made a few comments inline.

File: scripts/Package_BGA/bga_xilinx.yml (hunk 3 above)
Comment (herostrat) [reply scripts/Package_BGA/bga_xilinx.yml]: You are right.
I do not know why but I thought the versions with only one name were not included with the additional tags.
Will have this in mind!

**Comment by herostrat** [comment]
Thanks for the review, I fixed the problems.
I don't understand how the errors can still slip through.
I will take more care that there are no trivial ones in the future...

**Comment by evanshultz** [comment]
All is well. You're doing incredible work and thank you so much! More than two eyes give better results.

Push the updated footprints and then I'll merge them. :)

**Comment by herostrat** [comment]
Thank your very much.

The footprints are updated.

Do you want me to add the other footprints in one PR or do you prefer smaller chunks?

We are talking about additional ~19 for Kintex and Virtex and ~15 for the Zynq (although some of those are the same as in the Artix and Spartan series, where I will just update the description and tag with the additional information)

**Comment by evanshultz** [comment]
I'm happy if you make more script changes here by family, but then let's merge footprints in separate PRs. Does that work for you?

**Comment by herostrat** [comment]
Of course, I can split them all for better reviewability.
I just didn't want to spam you with PRs!

**Comment by herostrat** [comment]
Dammit I think I messed up my fork repository.
Is it still possible to merge it or do I have to make another PR for this?

**Comment by herostrat** [comment]
@pointhi The footprints were merged here https://github.com/KiCad/kicad-footprints/pull/1568

If @evanshultz and you have no objections, it would be awesome if this gets merged, so that I can continue without conflicts to add the remaining elements.

**Comment by herostrat** [comment]
Thanks a lot!

**Pull request merged**